src/Particles/babylon.particleSystem.ts

```diff
@@ -119,10 +122,33 @@
         private _actualFrame = 0;
         private _scaledUpdateSpeed: number;
 
-        constructor(public name: string, capacity: number, scene: Scene, customEffect?: Effect) {
+        // sheet animation
+        public startSpriteCellID = 0;
+        public endSpriteCellID = 0;
+        public spriteCellLoop = true;
+        public spriteCellChangeSpeed = 0; //(cellID++ every 4 frames). If this value is set to 0 (default) then we need to compute the speed base on lifetime like we are doing for colorDead. In other word, the particle will born with startCellId and die at endCellId
+
+        private _spriteCellWidth = 0;
+        private _spriteCellHeight = 0;
+        private _vertexBufferSize = 11;
+        // end of sheet animation
+
+        constructor(public name: string, capacity: number, scene: Scene, customEffect?: Effect, private spriteCellSize?: any, epsilon: number = 0.01) {

```

Comment:
here we should add a boolean to specify if the particles will rely on a sheet. this will give users the opportunity to change the sheet and the size of cells afterwards (by having a cellWidth and height as public properties)